sycl/tools/aie-intrinsic/aie-intrinsic.h

```diff
@@ -0,0 +1,74 @@
+#ifndef __AIE_INTRINSIC_HPP__
+#define __AIE_INTRINSIC_HPP__
+
+/// __SYCL_DEVICE_ONLY__ means we are on device code in SYCL
+/// __AIE_RT__ means we are compiling the device library in chess
+/// neither __SYCL_DEVICE_ONLY__ nor __AIE_RT__ means we are compiling for the
+/// host.
+
+#if defined(__SYCL_DEVICE_ONLY__) || defined(__AIE_RT__)
+/// When compiling for device the definition is in the library
+#define DECL_POSTFIX ;
+#if defined(__SYCL_DEVICE_ONLY__)
+/// When compiling for SYCL we need to mark function external or the compiler
+/// will complain about undefined function.
+#define DECL_PREFIX SYCL_EXTERNAL
+#else
+#define DECL_PREFIX
+#endif
+#else
+#include <cassert>
+/// When compiling for the host the functions need to compile because they will
+/// be compiled even they will not be executed.
+#define DECL_POSTFIX                                                           \
+  {                                                                            \
+    bool b = false;                                                            \
+    (void)b;                                                                   \
+    assert(b && "this should only be called on device");                       \
+    __builtin_unreachable();                                                   \
+  }
+#define DECL_PREFIX
+#endif
+
+#include <stdint.h>
+
+namespace aie_intr {
```

Comment:
```suggestion
namespace aie::intrinsics {
```
to avoid confusion with `interrupt` :-)